app/src/main/java/com/github/palFinderTeam/palfinder/ProfileViewModel.kt

```diff
@@ -94,4 +109,30 @@ class ProfileViewModel @Inject constructor(
         }
     }
 
+    /**
+     * We handle the block/unblock logic here to avoid handling coroutines in
+     * the adapter. Notice that we need to fetch again the user who wants to follow,
+     * that's because we need to have the more up to date info.
+     */
+    fun block(userId: String, otherId: String) {
```

Comment:
Also from what I see you can still be followed by & follow people you blocked.

Reply:
Ah nice catch will try to fix, there are other issues actually with the way the blocking work (in term of user experience) I might have to fix them for next sprint.